### PR TITLE
Bin executable, tmp dir based on OS, live results in browser

### LINKED ISSUE #1: Move to npx executable
We should move the current script into the bin folder and make it callable via `npx`, as this should bring a nice DX.
The only consideration is about the JSON output. Where should we put it?

### PR DESCRIPTION
**Pull request by buggyzap**
I have created a bin directive inside the package.json to execute the index.mjs as default npx command of the package #1 .

I have also made some code refactor and addition, like:

- Refactor the JSON results output, migrated from **results** folder and **tmp** to real **tmp** OS directory.
- I have created an ultra-simple http server that shows the JSON result after the generation and open the default browser to show that JSON

The user can choice to see the json or save it to a local file and rename it as they want.

I have tried to use native node.js packages to don't add any overhead.
Let me know if you like it.

PS. Sorry for the index.mjs syntax changes but my Prettier is configured differently, pls add the .prettierc file to have the same configuration between contributors.

### CODE REVIEW

File: index.mjs [hunk 1]
```diff
@@ -9,17 +13,17 @@ const repoUrl = process.argv[2];
...
 
-const tmpDir = path.join(import.meta.url.replace('file://', '').replace('index.mjs', ''), 'tmp');
-const resultsDir = path.join(import.meta.url.replace('file://', '').replace('index.mjs', ''), 'results');
+const tmpDir = os.tmpdir() + "tmp-spyone";
+
+const resultsDir = os.tmpdir() + "/results";
 
 if (!repoUrl || !daysAmount) {
-  console.error('Usage: node index.mjs <repo-url> <commit-count>');
+  console.error("Usage: node index.mjs <repo-url> <commit-count>");
```
Comment (Cadienvan) [inline index.mjs]: Shouldn't it be "Usage: npx @jointly/spyone <repo-url> <days-to-cover> <branch-name>"?
It was wrong even in the prev version :P

File: README.md [hunk 2]
```diff
@@ -1,29 +1,24 @@
 _spyone_ is a tool that generates a JSON heatmap of any Git repository's commits showing most edited files.  
 It comes from the italian word _spione_, which means _spy_.
 
-Please, keep in mind this project is not perfect. It was made to analyze our most edited files to create some sort of `heatmap` of the technical debt of our internal projects, so some manual actions are requested. This should be easily portable into an `npx` command and probably will in the future.
+Please, keep in mind this project is not perfect. It was made to analyze our most edited files to create some sort of `heatmap` of the technical debt of our internal projects.
 
 ## What does it do?
 
 This script downloads any Git repository and extracts statistics from its commits. The resulting data is then sorted by commit count, and if there is a tie, by the sum of additions and deletions in descending order. The result is saved in a folder called `results/` with a self-explanatory name (`yyyy-m-dd-hh-mm-{repo-slug}-{days-to-cover}-{branch-name}.json`).
```
Comment (Cadienvan) [inline README.md]: This should be changed to reflect the new behaviour

**Review by Cadienvan — CHANGES_REQUESTED** [review]

**Comment by Cadienvan** [comment]
Hi @buggyzap , this seems awesome!
Thank you a lot for improving our little project by all the tech team @ Jointly.

I just left a couple of request changes to reflect the new behaviour, I'll try your fork soon just to be sure everything is fine and then I'll open this PR up for review of the other maintainers!

File: index.mjs [hunk 3]
```diff
@@ -9,17 +13,17 @@ const repoUrl = process.argv[2];
...
 
 // Third argument passed via CLI is the name of the branch to consider
-const branchName = process.argv[4] || 'main';
+const branchName = process.argv[4] || "main";
 
-const tmpDir = path.join(import.meta.url.replace('file://', '').replace('index.mjs', ''), 'tmp');
-const resultsDir = path.join(import.meta.url.replace('file://', '').replace('index.mjs', ''), 'results');
+const tmpDir = os.tmpdir() + "tmp-spyone";
+
+const resultsDir = os.tmpdir() + "/results";
```
Comment (PellegrinoDurante) [inline index.mjs]: Missing a `/` before `tmp-spyone`?
Could be safer using `path.join()`?

File: index.mjs (hunk 1 above)
Comment (buggyzap) [reply index.mjs:23]: Yess 😆

**Comment by buggyzap** [comment]
@Cadienvan Hi guys do you think is better an interactive version with https://www.npmjs.com/package/enquirer or is an useless overhead and higher time of npx command execution?

**Review by PellegrinoDurante — APPROVED** [review]

**Comment by Cadienvan** [comment]
That was awesome @buggyzap , thank you so much for your contribution, I really liked these interactions.

I'll follow up with #4 and #5 as soon as I can, I promise!

I'll start by bringing this in and bump version!

**Pull request merged**